recipe-generator/src/main/java/ca/derekcormier/recipe/generator/TypescriptFilters.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.Map;
+
+import ca.derekcormier.recipe.cookbook.Cookbook;
+import ca.derekcormier.recipe.cookbook.CookbookUtils;
+import ca.derekcormier.recipe.cookbook.Primitive;
+import ca.derekcormier.recipe.cookbook.type.ArrayType;
+import ca.derekcormier.recipe.cookbook.type.EnumType;
+import ca.derekcormier.recipe.cookbook.type.FlagType;
+import ca.derekcormier.recipe.cookbook.type.ParamType;
+import ca.derekcormier.recipe.cookbook.type.PrimitiveType;
+import ca.derekcormier.recipe.cookbook.type.Type;
+import liqp.filters.Filter;
+
+public class TypescriptFilters {
+    public static Filter createTsTypeFilter(Cookbook cookbook) {
+        return new Filter("tstype") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                ParamType type = CookbookUtils.parseType(super.asString(value), cookbook);
+                return TypescriptFilters.toTsType(type);
+            }
+        };
+    }
+
+    public static Filter createTsParamFilter(Cookbook cookbook) {
+        return new Filter("tsparam") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                Map<String,String> param = (Map<String,String>)value;
+                String name = param.get("name");
+                ParamType type = CookbookUtils.parseType(param.get("type"), cookbook);
+
+                return (type.isVararg() ? "..." : "") + name + ": " + TypescriptFilters.toTsType(type);
+            }
+        };
+    }
+
+    public static Filter createJsParamFilter(Cookbook cookbook) {
+        return new Filter("jsparam") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                Map<String,String> param = (Map<String,String>)value;
+                String name = param.get("name");
+                ParamType type = CookbookUtils.parseType(param.get("type"), cookbook);
+
+                return (type.isVararg() ? "..." : "") + name;
+            }
+        };
+    }
+
+    public static Filter createTsValueFilter(Cookbook cookbook) {
+        return new Filter("tsvalue") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                String strType = super.asString(params[0]);
+
+                if (CookbookUtils.isKnownType(cookbook, strType)) {
+                    if (CookbookUtils.isPrimitiveType(strType)) {
+                        Primitive type = Primitive.fromAlias(strType);
+                        switch (type) {
+                            case BOOLEAN:
+                                return super.asString(value);
+                            case INTEGER:
+                                return super.asString(value);
+                            case FLOAT:
+                                return super.asString(value);
+                            case STRING:
+                                return "\"" + super.asString(value) + "\"";
```

Comment:
Does `super.asString(value)` need to be escaped? In case it has quotes in it...

Reply:
I think any strings inside would already be escaped.

Reply:
This is used for outputting the default value, which is specified in the cookbook, which should have escape characters already.

Reply:
On second thought...I'm not so sure. Let me write a test.